FEXCore/Source/Interface/Core/X86Tables/X86Tables.h

```diff
@@ -507,26 +507,24 @@ using U8U8InfoStruct = X86TablesInfoStruct<uint8_t>;
 using U16U8InfoStruct = X86TablesInfoStruct<uint16_t>;
 
 template<typename OpcodeType>
-static inline void GenerateTable(X86InstInfo *FinalTable, X86TablesInfoStruct<OpcodeType> const *LocalTable, size_t TableSize) {
+constexpr static inline void GenerateTable(X86InstInfo *FinalTable, X86TablesInfoStruct<OpcodeType> const *LocalTable, size_t TableSize) {
   for (size_t j = 0; j < TableSize; ++j) {
     X86TablesInfoStruct<OpcodeType> const &Op = LocalTable[j];
     auto OpNum = Op.first;
     X86InstInfo const &Info = Op.Info;
     for (uint32_t i = 0; i < Op.second; ++i) {
-      LOGMAN_THROW_AA_FMT(FinalTable[OpNum + i].Type == TYPE_UNKNOWN, "Duplicate Entry {}->{}", FinalTable[OpNum + i].Name, Info.Name);
```

Comment:
Not sure how I feel about dropping self-consistency checks for the sake of saving a few kilobytes of executable memory. How about re-adding the checks as a post=init verification step in `InitializeInfoTables`?

Luckily this can even be generically [as I've done here](https://github.com/FEX-Emu/FEX/compare/main...neobrain:FEX:opt_x86tables_init_next).

Reply:
Post initialization verification is hard to do since there will be table entries that remain having their type set to `TYPE_UNKNOWN`.
We're checking to ensure the values that we're setting are still `TYPE_UNKNOWN` before overwriting it. 
We can't do consteval asserts to match this behaviour as far as I can tell.

For your implementation of `CheckForInternalConflicts` it is implemented with the comment of `// TODO: Implement?`

Reply:
On second look I think `CheckForInternalConflicts` only ever intended to check for conflicts within each *individual* `LocalTable` but not between *different* once. The former makes the check trivial, hence the TODO. 

>  We can't do consteval asserts to match this behaviour as far as I can tell.

Yeah, that would require one `.cpp` file to have full visibility of all the various LocalTables, upon which it could concatenate them and then check for duplicates. And that's clearly contrary to the whole point of splitting these tables into different files in the first place.

I realized one thing though: Have you tried simply leaving the `LOGMAN_THROW_AA_FMT(...)` in? You should actually be able to have non-constexpr code in a `consteval` lambda as long as the code doesn't execute. It will only fail if the assert indeed is failing then. If for some reason it doesn't compile right now, I think there should be an easy way to make it work.

Reply:
Yes I tried leaving them in and it results in a compile failure.
```
/mnt/Work/Work/work/FEXNew/FEXCore/Source/Interface/Core/X86Tables/SecondaryGroupTables.cpp:15:80: error: call to consteval function 'FEXCore::X86Tables::(anonymous class)::operator()' is not a constant expression
std::array<X86InstInfo, MAX_INST_SECOND_GROUP_TABLE_SIZE> SecondInstGroupOps = []() consteval {
                                                                               ^
/mnt/Work/Work/work/FEXNew/FEXCore/Source/Interface/Core/X86Tables/X86Tables.h:516:7: note: non-constexpr function 'AFmt<const char *, const char *>' cannot be used in a constant expression
      LOGMAN_THROW_AA_FMT(FinalTable[OpNum + i].Type == TYPE_UNKNOWN, "Duplicate Entry {}->{}", FinalTable[OpNum + i].Name, Info.Name);
      ^
/mnt/Work/Work/work/FEXNew/FEXCore/include/FEXCore/Utils/LogManager.h:75:45: note: expanded from macro 'LOGMAN_THROW_AA_FMT'
#define LOGMAN_THROW_AA_FMT(pred, ...) do { LogMan::Throw::AFmt(pred, __VA_ARGS__); } while (0)
                                            ^
/mnt/Work/Work/work/FEXNew/FEXCore/Source/Interface/Core/X86Tables/SecondaryGroupTables.cpp:490:3: note: in call to 'GenerateTable(&Table._M_elems[0], &SecondaryExtensionOpTable[0], 384)'
  GenerateTable(&Table.at(0), SecondaryExtensionOpTable, std::size(SecondaryExtensionOpTable));
  ^
/mnt/Work/Work/work/FEXNew/FEXCore/Source/Interface/Core/X86Tables/SecondaryGroupTables.cpp:15:80: note: in call to '&[]() {
```

Reply:
Looks like it fails because that unconditionally calls `LogMan::Throw::AFmt`. What happens if you wrap the calls in the actual assertion condition?

```cpp
if (FinalTable[OpNum + i].Type != TYPE_UNKNOWN) {
      LOGMAN_THROW_AA_FMT(FinalTable[OpNum + i].Type == TYPE_UNKNOWN, "Duplicate Entry {}->{}", FinalTable[OpNum + i].Name, Info.Name);
}
```

(Obviously we might as well just use an `ERROR_AND_DIE_FMT` or similar at that point)

Reply:
Turns out that does work. Thought I tried that.